lib/aws/lex/conversation/support/mixins/responses.rb

```diff
@@ -8,51 +8,43 @@ module Mixins
           module Responses
             def close(opts = {})
               params = {
-                active_contexts: lex.active_contexts,
-                recent_intent_summary_view: pending_checkpoints,
-                session_attributes: lex.session_attributes
+                session_state: lex.session_state,
+                request_attributes: lex.request_attributes
               }.merge(opts)
               Response::Close.new(params).to_lex
             end
 
             def confirm_intent(opts = {})
               params = {
-                active_contexts: lex.active_contexts,
-                intent_name: lex.current_intent.name,
-                recent_intent_summary_view: pending_checkpoints,
-                session_attributes: lex.session_attributes,
-                slots: lex.current_intent.slots
+                session_state: lex.session_state,
+                request_attributes: lex.request_attributes
               }.merge(opts)
               Response::ConfirmIntent.new(params).to_lex
             end
 
             def delegate(opts = {})
               params = {
-                active_contexts: lex.active_contexts,
-                recent_intent_summary_view: pending_checkpoints,
-                session_attributes: lex.session_attributes,
-                slots: lex.current_intent.slots
+                session_state: lex.session_state,
+                request_attributes: lex.request_attributes
               }.merge(opts)
               Response::Delegate.new(params).to_lex
             end
 
             def elicit_intent(opts = {})
               params = {
-                active_contexts: lex.active_contexts,
-                recent_intent_summary_view: pending_checkpoints,
-                session_attributes: lex.session_attributes
+                session_state: lex.session_state,
+                request_attributes: lex.request_attributes
               }.merge(opts)
               Response::ElicitIntent.new(params).to_lex
             end
 
             def elicit_slot(opts = {})
               params = {
-                active_contexts: lex.active_contexts,
-                intent_name: lex.current_intent.name,
-                recent_intent_summary_view: pending_checkpoints,
-                session_attributes: lex.session_attributes,
-                slots: lex.current_intent.slots
+                session_state: lex.session_state,
+                request_attributes: lex.request_attributes,
+                intent_name: lex.session_state.intent.name
               }.merge(opts)
+              lex.session_state.intent = lex.intents.find { |i| i.name == params.fetch(:intent_name) }
```

Comment:
The `find` call may return `nil` here if the user messes up the input `intent_name` parameter.

Do you think it's worth a quick guard clause/raise in this case?

Reply:
Apparently it's possible to pass a callable instance as an argument to `Enumberable#find` and it'll get called in the case nothing is found:

```ruby
[].find(-> { raise ArgumentError, 'not found' }) { |e| e == 2 }
```

Reply:
good suggestion